src/ssl.c

```diff
@@ -12286,7 +12288,27 @@ int AddSession(WOLFSSL* ssl)
             return BAD_MUTEX_E;
         }
 
-        idx = SessionCache[row].nextIdx++;
+        for (i=0; i<SESSIONS_PER_ROW; i++) {
+            if (ssl->options.tls1_3) {
```

Comment:
Same code regardless of tls1_3???

Reply:
Not the same. For TLS 1.3 the session is stored in `ssl->session` while for other protocols `ssl->arrays` is used.